### ISSUE TITLE
Compile Issue: Failure to Load Core or Content after recent commits (Armv7/Armv8)

### ISSUE THREAD

**Issue by KMFDManic** [libretro/libretro-uae#189]
Not sure exactly where things got broken.  But, I have been able to easily compile and have the Core run, since you guys (@rsn8887 and @sonninnos ) have done an absolutely incredible job fixing things up.  But, something recent has broken things on "our end" platform wise, for the Mini Classics, which use Armv7/Armv8.  Here is a log that another great mate in the Scene, @bslenul,  grabbed whilst confirming things didn't load for him on my Compile, either:

https://pastebin.com/n6StZppf

Thank you both so much, and truly hoping this can easily be pinpointed, as I would absolutely love to get the Amiga CD32 up and running, amongst other great recent additions!

**Comment by sonninnos** [comment]
I'm afraid that log gives no useful information, because for now UAE spits all verbosity to stderr.

Build log would be useful too.

**Comment by KMFDManic** [comment]
Alright @sonninnos gonna get back onto Ubuntu, and do a rebuild from scratch and show you the build log.  Thanks so much!  P-UAE is in my top tier Cores, along with MAME 2003:)

**Comment by KMFDManic** [comment]
Alright @sonninnos Here ya go!  Looking forward to getting things fixed up!

https://pastebin.com/ncrbTsfT

**Comment by sonninnos** [comment]
I don't know where to begin. I would need a log of a build that does not build.

RA log shows no apparent errors either, but a state that will never work. It is trying to boot a HDF with A500 KS1.3, and it is not possible. If there is no "Automatic" option for the model (which would select A600 with HDFs), then the version is too old anyway.

And if the HDF is a WHDLoad version, and not a standalone HDF, it should try to open WHDLoad.hdf before Agony.hdf. So if that is missing too, it also will not work.

**Comment by rsn8887** [comment]
Is it only a problem when you compile it yourself? Did you try the buildbot built for mini (if there is one, is there one?)

**Comment by rsn8887** [comment]
As far as I know, minis use different codebase than master. Did you check you are actually compiling the latest master from github.com/libretro/libretro-uae ? You can check the commit Id in the Retroarch menu on bottom left. It should match latest commit on official master branch. If it doesn’t there’s no need to debug anything here. You would have to talk to the mini repo maintainers.

**Comment by KMFDManic** [comment]
@sonninnos @rsn8887 I have been able to properly compile, load, and run content without issue, up til the last few weeks worth of commits.  Usually, if I do have a compile issue, I might test a build bot version, as well, and they tend to have the same issues.  A few other Cores have needed a few fixes here and there to account for such variables.  BlueMSX, Flycast, Reicast, 4DO, Yabause, etc.  In this case, everything runs flawlessly, up til a few weeks ago.  I compile roughly 100 Cores on a regular basis, and have generally been able to get any issues quickly resolved, whether on my end, or elsewhere...especially with the feedback of other developers, such as you two:)  

As far as Agony running, HDF wise, the game runs fine on the older builds from a few weeks ago.  We run base model at 600, which works great for a majority of games.  I even recompiled without the current commits, and everything worked great.  Sometimes, I have built, had the Core not work, then just rebuilt on the next commit, and things were fine.  

I just verified that I am using the latest master, as the exact commit shows at bottom left, as asked about, above.  So, unless it can be pinpointed, I may just have to recompile and try to revert a few commits and see where I can successfully have a working build again.  But, thanks so much for both of you helping with this.  Truly means a lot.

**Comment by rsn8887** [comment]
I am not sure what build @bslenul is running, but it is not build from any of our sources. His commit is 0186c1b. From the log you posted: 
```P-UAE 2.6.1
Git Commit: 0186c1b16f7181ffa02d73e6920d3180ce457c46
Build date: Dec 22 2019 22:49:37``

Our latest commit e953456. His commit is not even in our commit history!

**Comment by KMFDManic** [comment]
@rsn8887 The e953456 commit is in the build I provided @bslenul.  The log he gave may be pointing to RetroArch Build we are using.  I tested this on a few different platforms, to verify that it didn't work there, as well.  We have nearly always been able to sort out issues, when they do arise.  I don't mind recompiling, if I need to.  I am sure we will get to the bottom of this.  Oddly enough, when Cores do tend to crash, if you go to history, the games you attempted to load will not usually be listed in history.  This time, however, they do show in history.

**Comment by rsn8887** [comment]
Can you bisect and find the core commit that broke things for you?

**Comment by rsn8887** [comment]
You could also try using -UCD32 and -UGARY in platform flags for minis (in the makefile) to disable CD32 and GARY chip support, just to narrow down the issue a bit.

**Comment by KMFDManic** [comment]
No, on P-UAE, I did exact build of the Master, to verify it works.  I used your Repo.  For RetroArch, also using exact Master for that.  The Cores like MAME 2003 Xtreme, I just linked to the exact commits I build off of for that.  And, to determine where P-UAE breaks, I would need to go back a few commits to see.  I JUST did a video using the last working build, which is fantastic:

https://www.youtube.com/watch?v=0BNaX1cBfBY (27 minute point)  Walker is a fun game!

I will try the other flags, and see if that helps!  Thank you, @rsn8887 :)

**Comment by rsn8887** [comment]
Your log shows you are compiling `platform=rpi` but that is not a mini classic platform. AFAIK mini classic would use `platform=classic_armv7_a7` . So you are saying rpi build is broken? Your original post seemed to indicate the classic mini platforms are broken.

**Comment by rsn8887** [comment]
- You might try to rename your core config file for PUAE and start with a fresh default config.
- You could try to add the flag "-DUSE_NAMED_SEMAPHORES" in PLATFLAGS for your platform. It fixed OSX, which broke after CD32 commits.

**Comment by rsn8887** [comment]
Another thought: On your platform, you get the modern core options, right? E.g. when you start a core that supports v2 options, like Dosbox, Vice, or PUAE, and go to quickmenu->core options, you see the small print description under each option?

**Comment by KMFDManic** [comment]
@sonninnos @rsn8887 

I thought about the Core Options being a potential issue.  But, Start Core also crashes.  I will try running with defaults, again.  The commit of the Core that does work in my video is:

d38f517  

I usually recompile every single new commit.  But, I had been working extra Overtime at work, and fell a few weeks behind.

And, yes, armv7 works on retropie and vice versa.  So, this break is probably broken on retropie, as well.

**Comment by sonninnos** [comment]
This does not work either?
https://github.com/libretro/libretro-uae/commit/0a85b4fc4f1ad58c02bbc6592e03bc8a9c9aefa1

And this?
https://github.com/libretro/libretro-uae/commit/80068b51b433805923748f19c054c7157a9389ca

**Comment by KMFDManic** [comment]
@sonninnos d38f517 was the last commit I built with due to work schedule being more hectic than normal.  I would need to recompile to try any commits after that to pinpoint the exact one where it broke.  But, I can try the flags suggested above, as well.

**Comment by KMFDManic** [comment]
Gonna try the above suggestions, and see if I can get a successful working build, then respond.

**Comment by KMFDManic** [comment]
@sonninnos @rsn8887 Thank you both so much for your feedback and suggestions.  We got things fixed up on our end, and I will post the .diff in the AM, which may help you two adjust the Makefile to avoid other potential and similarly minded Platform conflicts.  I had zero issue compiling up til a few weeks ago.  But, now things should be fine, again:)  I will do another follow-up response, and we can go from there.

**Comment by KMFDManic** [comment]
@sonninnos @rsn8887 

This fixed the compilation for armv7/rpi

https://pastebin.com/u/KMFDManic

Great friend, @madmonkey1907 happened to check in on things since he had been following my Repos, and helped with the fix.  Thanks again, looking forward to .chd implementation.  If either of you have difficulties with .chd, I recommend pinging @fabrice-martinez who did an exceptional job with neo-cd and implementing .chd support for that!

https://github.com/libretro/neocd_libretro

Closing the issue now, but will look for any responses, and act accordingly!  Again, thanks so much for both of you making one of the hardest to work on Cores better and better each and every commit Update:)

**Comment by rsn8887** [comment]
Ok, should be fixed in commit 5a7e4a9f77be8
